### PR TITLE
#167165067 User Should Be Able To Follow, Unfollow, View Followers And Following.

### PR DESCRIPTION
**Pull request by Tunji17**
#What does this PR do?
Adds follow functionality to users

#### Description of Task to be completed?
Adds user can follow other users route
Adds user can unfollow other users route
Adds user can get a users followers routes
Adds user can get a users followings routes
Refactor model files to account for UUIDs change
Adds documentation for the 4 routes
Adds integration tests for the 4 routes
Adds input validator for specifically checking id params

#### Any background context you want to provide?
Only authenticated users should be able to follow and or unfollow other users, I need a view to manage those I follow and be able to see those who follow me

#### What are the relevant pivotal tracker stories?
[#167165067](https://www.pivotaltracker.com/story/show/167165067)

#### Screenshots (if appropriate)
Screenshot 2019-08-02 at 9 44 13 AM
Screenshot 2019-08-02 at 9 42 48 AM
Screenshot 2019-08-02 at 9 42 38 AM

### CODE REVIEW

**Review by mekzy-o — COMMENTED** [review]
Nice Job Tunji. LGTM. Try to keep minimal file changes. You can always break down your task

**Review by mekzy-o — APPROVED** [review]

**Comment by Tunji17** [comment]
> Nice Job Tunji. LGTM. Try to keep minimal file changes. You can always break down your task

Thank you for the feedback, i'll implement that next time

File: controllers/followershipController.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    if (id === userid) {
+      return res.status(400).send({
+        errors: {
+          message: 'You are not allowed to follow yourself',
+        }
+      });
+    }
+
+    await Follower.findOrCreate({ where: { followerId: userid, userId: id } })
+      // eslint-disable-next-line no-unused-vars
```
Comment (mazma1) [inline controllers/followershipController.js]: Why is this here?

File: controllers/followershipController.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      include: [
+        {
+          model: User,
+          as: 'followings',
+          attributes: ['id', 'firstName', 'lastName']
+        }
+      ]
+    });
+
+    return res.status(200).send({
```
Comment (mazma1) [inline controllers/followershipController.js]: Why don't you use the `ServerResponse` module in the codebase?

File: controllers/followershipController.js (hunk 1 above)
Comment (Tunji17) [reply controllers/followershipController.js]: there's a follow argument that i'm not using in the then block that returns the followers id and users id, and i left it there incase someone may need it in future

File: controllers/followershipController.js (hunk 2 above)
Comment (Tunji17) [reply controllers/followershipController.js]: the serverResponse module has a data object that is not in the documentation, and I saw some other controllers using it so I picked up on it

Comment (mazma1) [reply controllers/followershipController.js]: I don't understand this explanation 😏

File: controllers/followershipController.js (hunk 1 above)
Comment (mazma1) [reply controllers/followershipController.js]: You can take it out, the future will take care of itself

File: controllers/followershipController.js (hunk 2 above)
Comment (Tunji17) [reply controllers/followershipController.js]: do you think I should change it to the server response? so I can change it after rebasing before I push again

Comment (mazma1) [reply controllers/followershipController.js]: What do you think? Do you think you should change it? And why?

I'm assuming we are on the same page as per what I pointed out

Comment (Tunji17) [reply controllers/followershipController.js]: Yes we are, it's one of those conventions we just overlooked as a team, maybe I'll talk to everyone and I'll get back to you

File: controllers/AuthController.js [hunk 3]
```diff
@@ -285,11 +280,12 @@ export default class AuthController {
...
-    return res.status(200).send({
-      email,
-      token,
-      message: 'Your reset link has been sent to your email'
-    });
+    // return res.status(200).send({
+    //   email,
+    //   token,
+    //   message: 'Your reset link has been sent to your email'
+    // });
```
Comment (mazma1) [inline controllers/AuthController.js]: What did you leave this here?

File: controllers/followershipController.js (hunk 1 above)
Comment (mazma1) [reply controllers/followershipController.js]: Have you taken out this "unused-var"?

Comment (Tunji17) [reply controllers/followershipController.js]: yes i have

File: controllers/AuthController.js (hunk 3 above)
Comment (mazma1) [reply controllers/AuthController.js]: @Tunji17 👆🏽

Comment (Tunji17) [reply controllers/AuthController.js]: Sorry it was an oversight, i've removed it @mazma1
